canister/src/lib.rs

```diff
@@ -93,18 +94,21 @@ pub fn get_current_fee_percentiles(
 ) -> Vec<MillisatoshiPerByte> {
     verify_api_access();
     verify_network(request.network.into());
+    verify_fully_synced();
```

Comment:
I believe we're not testing any of these endpoints that this check is being applied. Maybe we can an add an e2e test that puts the bitcoin canister in a state where it's not synced, and verifies that these API requests get rejected?

Reply:
I am not sure how beneficial that will be, because it will require some time to make such tests I am not sure I have any idea how I will put the canister in a not synced state without too much work. And we are already testing the behavior of verify_fully_synced(), hence the only way to make it not work is to delete the check from the methods.  But, if you feel strongly about it, we can add it to the follow-up. This PR is already too long:)

Reply:
But testing of `get_current_fee_percentiles`, `get_balance`, and `get_utxos` should be easy. Do you think that will be enough?

Reply:
Here's how I think about it:

1. We're adding a new functionality.
2. This functionality, if it's buggy, poses a significant risk.

By the above logic, to me this is already a very strong reason to add test cases. Note that in my logic, I didn't consider how difficult writing the test would be nor how simple the implementation is. These, IMHO, are not the right problems to decide on whether tests should be included or not. If writing tests is painful, that means we need to improve our tooling to make testing easier. And if an implementation is simple enough today, it is not a guarantee that the implementation won't evolve to be more complex in the future.

> I am not sure I have any idea how I will put the canister in a not synced state without too much work.\
It's actually very simple. If you create something like the e2e scenarios where you are returning no blocks and a few `next` block hashes forever, the bitcoin canister will be in a state where it's not synced.

I'm ok with that to be part of a follow-up, but let's create a JIRA ticket for it please so that we don't forget.

Reply:
[Here](https://dfinity.atlassian.net/browse/EXC-1362) is jira link.